reporters/dogstatsd.go

```diff
@@ -74,8 +83,24 @@ func NewDogStatsD(config *viper.Viper, logger *logrus.Logger) (*DogStatsD, error
 	if err != nil {
 		return nil, err
 	}
+
+	restartTimeout := config.GetDuration("reporters.dogstatsd.restartTimeout")
 	r := config.GetString("reporters.dogstatsd.region")
-	dogstatsdR := &DogStatsD{client: c, region: r}
+
+	dogstatsdR := &DogStatsD{
+		client:       c,
+		statsdClient: c.Client.(*statsd.Client),
```

Comment:
What's the difference between client and statsdClient?

Reply:
The `c.Client` is an interface from [this library](https://github.com/topfreegames/extensions). It abstracts the original implementation of Datadog's client.

While `statsdClient` is the casted version of `c.Client` to `statsd.Client`. I needed cast to be able to access the `Close` method.